modules/packages/BLAS.chpl

```diff
@@ -695,8 +695,662 @@ module BLAS {
     return _ldA;
   }
 
+  /* Level 1 BLAS */
 
-  /*
+/*
+    Wrapper for the `ROTG routines <http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_ga2f65d66137ddaeb7ae93fcc4902de3fc.html#ga2f65d66137ddaeb7ae93fcc4902de3fc>`_
+
+    Given the Cartesian coordinates ``(a, b)`` of a point ``p``, these routines
+    return the parameters ``a``, ``b``, ``c``, and ``s`` associated with the
+    `Givens rotation <https://en.wikipedia.org/wiki/Givens_rotation>`_ that
+    zeros the y-coordinate of the point::
+
+      | c  s |   |a|   |r|
+      |      | . | | = | |
+      |-s  c |   |b|   |0|
+
+    Input Parameters: ``a``, ``b``
+
+    Output Parameters:
+
+    - ``r`` stored in ``a`` parameter: length vector of ``(a, b)``
+    - ``z`` stored in ``b`` parameter: (stored in ``a`` parameter), z (stored in b
```

Comment:
I'm having a hard time understanding what you're trying to say for ``z`` here. Also, should ``c`` and ``s`` be on their own lines?